bin/docker-cmd.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+# Always generate a random cookie unless set explicitly
+if [ -z "${RELEASE_COOKIE}" ]; then
+  export RELEASE_COOKIE="$(openssl rand -base64 42)"
+else
+  export RELEASE_COOKIE
+fi
```

Comment:
Instead of defining this here, we can do so in the `rel/env.sh.eex` file (run `mix release.init` to generate it) - which is useful because it works even if we are not running the release via Docker.

Reply:
Wait a second.... if we define it randomly then it is impossible to connect to the node via the shell. Maybe that's ok with Docker but we don't want that as part of the release... so please ignore the comment above.

I assume that if we publish those images, the default value of RELEASE_COOKIE will be hardcoded inside the image and shared with everyone? Maybe we should even remove the RELEASE_COOKIE file...

Reply:
The default value (in the cookie file) is not gonna be used though, because either the user provides `RELEASE_COOKIE` as an env variable when running the container, or we generate it randomly *on every container startup*. That's the desired behaviour, right?

Reply:
That's exactly why we set `RELEASE_COOKIE` in this startup script, and not as `ENV` as part of the image definition.

Reply:
Yes, yes. I am thinking we should remove it when assembling the image just so it fails in case it is used by accident... another option is to introduce a LIVEBOOK_COOKIE and we always override the RELEASE_COOKIE one. On one side, we would keep the env vars all consistent, on the other side I am not sure if we want to allow LIVEBOOK_COOKIE everywhere? Would it be useful for the escript mode?

Reply:
Good question. So one usage could be for the attached runtime, because one option would be to specify it along with the attached node name, but if we dynamically change current node cookie won't this break the current connections? If that's the case, having a global option when running the notebook sounds better, and so it would make sense to have it for escript as well. And consequently naming the env var LIVEBOOK_COOKIE makes much sense. My only question would be how should we configure the cookie at runtime? Just set it before application startup?

Reply:
I think we can set it on app start before we start the distribution? 🤔

Reply:
The node has to be alive for us to set the cookie, so perhaps after we ensure the distribution is enabled?

Reply:
I've just added `LIVEBOOK_COOKIE` and `--cookie`, both work as expected with Escript and Docker. If none is specified we always generate one randomly, which also keeps things consistent :>

Reply:
This should close #234, right? Given the cookie is global and not per-connection that's probably the best we can do.

Reply:
No, #234 is still valid because we can set the cookie of individual nodes too: http://erlang.org/doc/man/erlang.html#set_cookie-2

Reply:
Ah, so if we do `:erlang.set_cookie(:some@node, :teacup)`, this means we will use `:teacup` as the cookie when connecting to `:some@node`, but our primary cookie is not affected?

Reply:
Correct.

Reply:
Perfect!